lib/elixir/test/elixir/kernel/warning_test.exs

```diff
@@ -1786,6 +1786,17 @@ defmodule Kernel.WarningTest do
              "extra parentheses on a remote function capture &System.pid()/0 have been deprecated. Please remove the parentheses: &System.pid/0"
   end
 
+  test "\"with\" warns when given an argument which is not a match" do
+    assert capture_err(fn ->
+             Code.eval_string("""
+                  with fn -> true end do
+                    :warn
+                  end
```

Comment:
Can we indent this at the same level as the call to `Code.eval_string/1`? Not a pressing issue but would be nicer :)